api/integrations/slack/views.py

```diff
@@ -0,0 +1,94 @@
+import uuid
+
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from django.shortcuts import redirect
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from slack_sdk.oauth import AuthorizeUrlGenerator
+
+from integrations.common.views import IntegrationCommonViewSet
+from integrations.slack.models import SlackConfiguration, SlackEnvironment
+from integrations.slack.serializers import (
+    SlackChannelListSerializer,
+    SlackEnvironmentSerializer,
+)
+from integrations.slack.slack import SlackWrapper
+
+from .exceptions import InvalidStateError
+
+
+class SlackEnvironmentViewSet(IntegrationCommonViewSet):
+    serializer_class = SlackEnvironmentSerializer
+    pagination_class = None  # set here to ensure documentation is correct
+    model_class = SlackEnvironment
+
+    @action(
+        detail=False,
+        methods=["GET"],
+        url_path="channels",
+        serializer_class=SlackChannelListSerializer,
+    )
+    def get_channels(self, request, *args, **kwargs):
+        env = self.get_environment_from_request()
+
+        try:
+            config = SlackConfiguration.objects.get(project=env.project)
+        except ObjectDoesNotExist:
+            return Response(
+                "Slack api token not found. Please generate the token using oauth",
+                status.HTTP_400_BAD_REQUEST,
+            )
+        slack_wrapper = SlackWrapper(api_token=config.api_token)
+        serializer = self.get_serializer(
+            data=slack_wrapper.get_channels_data(), many=True
+        )
+        serializer.is_valid()
+        return Response(serializer.data)
+
+    @action(detail=False, methods=["GET"], url_path="callback", permission_classes=[])
+    def slack_oauth_callback(self, request, environment_api_key):
+        code = request.GET.get("code")
+        if not code:
+            return Response(
+                "code not found in query params",
+                status.HTTP_400_BAD_REQUEST,
+            )
+        env = self.get_environment_from_request()
+        validate_state(request.GET.get("state"), request)
+        bot_token = SlackWrapper(
+            code, self._get_slack_callback_url(environment_api_key)
```

Comment:
I'm not sure why we're seemingly using a callback URL as a channel ID here?

Reply:
fixed